src/OpenLoco/src/Ui/ViewportInteraction.cpp

```diff
@@ -1194,7 +1194,7 @@ namespace OpenLoco::Ui::ViewportInteraction
         args.index = track->sequenceIndex();
         args.trackObjType = track->trackObjectId();
         args.type = 1U << bh;
-        args.modSection = 0;
+        args.modSection = Windows::Construction::getLastSelectedTrackModSection();
```

Comment:
Just thinking about this we should probably make this dependent on the window being open on L1187. (Although i know you can't right click interact with track mods when the window isn't open so its not like we can currently end up in a bad situation)

Reply:
That would be future-proofing it, I suppose. But indeed, these two `rightReleased` functions are both already checking for the construction window being open.

I could have `getLastSelectedTrackModSection` return 0 if the construction window isn't open?

Reply:
yeah that would also work